### PR TITLE
Add a cts to be able to cancel the application task

### PR DESCRIPTION
**Pull request by vicancy**
Add a cts to be able to cancel the application task if the user's application code is an endless one

### CODE REVIEW

File: src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs [hunk 1]
```diff
@@ -307,14 +307,57 @@ private void AddClientConnection(ClientConnectionContext connection, OpenConnect
...
+            var task = await Task.WhenAny(app, cancelTask);
+
+            if (task != app)
+            {
+                // cancel the application task, to end the outgoing task
+                connection.Application.Input.CancelPendingRead();
+                Log.ApplicationTaskCancelled(Logger);
+            }
+
+            await task;
```
Comment (terencefan) [inline src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs]: Why `await task` again? Should we wait for `app` instead?

Comment (vicancy) [reply src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs]: await task for the task to end or throw

File: src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.Log.cs [hunk 2]
```diff
@@ -44,8 +44,8 @@ private static class Log
             private static readonly Action<ILogger, string, Exception> _serviceConnectionConnected =
                 LoggerMessage.Define<string>(LogLevel.Debug, new EventId(20, "ServiceConnectionConnected"), "Service connection {ServiceConnectionId} connected.");
 
-            private static readonly Action<ILogger, Exception> _applicationTaskTimedOut =
-                LoggerMessage.Define(LogLevel.Error, new EventId(21, "ApplicationTaskTimedOut"), "Timed out waiting for the application task to complete.");
+            private static readonly Action<ILogger, Exception> _applicationTaskCancelled =
+                LoggerMessage.Define(LogLevel.Error, new EventId(21, "ApplicationTaskCancelled"), "Cancelled running application code, probably caused by time out.");
```
Comment (BrennanConroy) [inline src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.Log.cs]: nit: Canceled

Comment (vicancy) [reply src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.Log.cs]: According to https://www.grammarly.com/blog/canceled-vs-cancelled/https://grammar.yourdictionary.com/grammar/spelling-and-word-lists/cancelled-or-canceled-what-s-the-correct-spelling.html, both `canceled` and `cancelled` are correct? 😃

Comment (BrennanConroy) [reply src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.Log.cs]: Yeah, but American English is preferred over British English when exposing logs and APIs.

Couple examples:
https://source.dot.net/#System.Private.CoreLib/Task.cs,c0742f865ee9be11,references
https://source.dot.net/#System.Private.CoreLib/Task.cs,3dc5e4fb145ac32f
https://source.dot.net/#System.Private.CoreLib/OperationCanceledException.cs,a0a8b79c277cf327

Comment (vicancy) [reply src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.Log.cs]: Thanks for sharing! will update

**Pull request merged**